src/main.rs

```diff
@@ -1,25 +1,34 @@
-extern crate requests;
-extern crate bigdecimal;
-extern crate json;
+extern crate reqwest;
+extern crate rust_decimal;
+extern crate serde;
 
-// adds method .json() to the Response class
-use requests::ToJson;
+use rust_decimal::Decimal;
+use serde::Deserialize;
 
-use std::str::FromStr;
-use bigdecimal::BigDecimal;
+#[derive(Deserialize)]
+#[serde(rename_all = "UPPERCASE")]
+struct Rates {
+    // btc: Decimal,
+    eth: Decimal,
+}
 
-//use requests::{Codes, Request, Response, StatusCode};
+#[derive(Deserialize)]
+struct Data {
+    // currency: String,
```

Comment:
It's nice that we don't have to exhaustively define the structure when we only need part of it. In fact, the compiler complains if you do!